src/v2i-hub/CARMAStreetsPlugin/src/J2735MapToJsonConverter.cpp

```diff
@@ -0,0 +1,267 @@
+
+
+#include "J2735MapToJsonConverter.h"
+
+namespace CARMAStreetsPlugin
+{
+
+    void J2735MapToJsonConverter::convertJ2735MAPToMapJSON(const std::shared_ptr<MapData> mapMsgPtr, Json::Value &mapJson) const
+    {
+        // Construct metadata
+        Json::Value metadata;
+        auto timestamp_utc = std::chrono::duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+        metadata["timestamp"] = std::to_string(timestamp_utc);
+        mapJson["metadata"] = metadata;
+
+        // Construct Map Data
+        Json::Value mapDataJson;
+        mapDataJson["layer_id"] = mapMsgPtr->layerID;
+        mapDataJson["msg_issue_revision"] = std::to_string(mapMsgPtr->msgIssueRevision);
+        mapDataJson["layer_type"] = std::to_string(*mapMsgPtr->layerType);
+
+        // Construct intersections
+        const IntersectionGeometryList *intersections = mapMsgPtr->intersections;
+        Json::Value intersectionsJson;
+        // Assume there is only one intersection geometry for each intersection
+        for (size_t i = 0; i < intersections->list.count; i++)
+        {
+            Json::Value intersectionJson;
+            if (intersections->list.array != nullptr)
+            {
+                auto intersection = intersections->list.array[i];
+                intersectionJson["id"]["id"] = std::to_string(intersection->id.id);
```

Comment:
What element does id id reference. Does the MAP message have an id and then include intersection id's as well. Is there a sample XML you could send me or include in some documentation as reference to what the JSON structure of outgoing map messages will be.

Reply:
I can send you XML offline